src/main/java/io/cdap/plugin/servicenow/ServiceNowBaseConfig.java

```diff
@@ -146,8 +147,8 @@ public void validateTable(String tableName, SourceValueType valueType, FailureCo
         // Removed config property as in case of MultiSource, only first table error was populating.
         collector.addFailure("Table: " + tableName + " is empty.", "");
       }
-    } catch (OAuthSystemException | OAuthProblemException e) {
-      collector.addFailure("Unable to connect to ServiceNow Instance.",
+    } catch (Exception e) {
+      collector.addFailure(String.format("Unable to connect to ServiceNow Instance. Cause: %s", e.getStackTrace()),
```

Comment:
Don't add stackTrace as a string as it will make the error message very hard to hard and it is not useful for the customer.